server/src/watchdog.js

```diff
@@ -138,3 +149,45 @@ exports.submit = function(shot) {
     mozlog.error("watchdog-failed-submission", {err: e});
   });
 };
+
+exports.handleResult = function(req) {
+  getSubmission(req.params.submissionId).then(record => {
+    if (record.request_id !== req.body.watchdog_id) {
+      throw new Error(`Received mismatching Watchdog ID in callback for submission ${record.id}.`);
+    }
+    if (req.query.nonce !== record.nonce) {
+      throw new Error(`Received mismatching nonce in callback for submission ${record.id}.`);
+    }
+    if (record.positive_result !== null) {
+      throw new Error(`Watchdog submission ${record.id} already has a result.`);
+    }
+
+    if (req.body.positive === false) {
```

Comment:
`if (req.body.positive)` ? Unless there's some weird in-between where it's null...?